src/core/unit.c

```diff
@@ -1340,13 +1340,29 @@ const char *unit_description(Unit *u) {
         return strna(u->id);
 }
 
-const char *unit_status_string(Unit *u) {
+const char* unit_status_string(Unit *u, char **combined) {
```

Comment:
the semantics of the `combined` parameter are weird. It's a strict output parameter, but sometimes it's not initialized. That's icky.

At the very least it should be initialized to NULL if not used. And this deserves a clear comment that it' sometimes a pointer and sometimes not.

Maybe rename `ret_combined_buffer` or so, indicating this doesn't return te combined string really, but just the buffer for it, if needed.